temporalio/lib/temporalio/worker_versioning.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Temporalio
+  WorkerDeploymentVersion = Data.define(
```

Comment:
Can we change this filename to `worker_deployment_version.rb`? While we have a rare file or two that encompass multiple top-level things not relating to the filename, _most_ files should be named after the single module or single class they define/represent.

If we think there may be multiple classes representing versioning, we should consider a versioning module